client/my-sites/hosting/staging-site-card/index.js

```diff
@@ -81,26 +93,58 @@ export const StagingSiteCard = ( {
 	}, [ dispatch, loadingError ] );
 
 	const stagingSite = useMemo( () => {
-		return stagingSites && stagingSites.length ? stagingSites[ 0 ] : [];
+		return stagingSites?.length ? stagingSites[ 0 ] : {};
 	}, [ stagingSites ] );
-	const hasSiteAccess = ! stagingSite.id || Boolean( stagingSite?.user_has_permission );
 
-	const showAddStagingSite =
-		! isLoadingStagingSites && ! isLoadingQuotaValidation && stagingSites?.length === 0;
-	const showManageStagingSite =
-		! isLoadingStagingSites && ! isLoadingQuotaValidation && stagingSites?.length > 0;
+	const { data: lock, isLoading: isLoadingLockQuery } = useGetLockQuery( siteId, {
+		enabled: ! disabled || !! stagingSite.id,
+		refetchInterval: 5000,
+		onError: () => {
+			setIsErrorValidQuota( true );
+		},
+	} );
+
+	const hasCompletedInitialLoading =
+		! isLoadingStagingSites && ! isLoadingQuotaValidation && ! isLoadingLockQuery;
+
+	const stagingSiteStatus = useSelector( ( state ) => getStagingSiteStatus( state, siteId ) );
+	const transferStatus = useCheckStagingSiteStatus( stagingSite.id, hasCompletedInitialLoading );
+
+	const isStagingSiteTransferComplete = useMemo( () => {
+		return (
+			stagingSiteStatus === StagingSiteStatus.COMPLETE ||
+			stagingSiteStatus === StagingSiteStatus.REVERTED ||
+			stagingSiteStatus === StagingSiteStatus.NONE ||
+			stagingSiteStatus === StagingSiteStatus.UNSET
+		);
+	}, [ stagingSiteStatus ] );
```

Comment:
We could move this code into a new selector. No need to useMemo, since it's not a heavy computation.